### ISSUE TITLE
Can't connect to db not supporting SSL

### ISSUE THREAD

**Issue by jacek-wojciechowski-dell** [manywho/service-identity#3]
@jonjomckay 

I've tried the latest version of this service and it doesn't work for me, no matter if "Database SSL?" is set or not.

In case it's set: org.postgresql.util.PSQLException: The server does not support SSL." - which is understandable and OK

In case it's not set: org.postgresql.util.PSQLException: Invalid sslmode value: prefer" -
shouldn't this be something else that will not fail in case SSL is not supported?

I think it's an important use case as most people that play with this have default Postgres config which does not have SSL

**Comment by jonjomckay** [comment]
@jacek-wojciechowski-dell hmmm dafuq - it looks like the PostgreSQL JDBC driver [doesn't support `prefer`](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java#L99)?!

I guess we should use `disable` then? Which isn't ideal...

**Comment by jonjomckay** [comment]
P.S. you can see the modes PostgreSQL supports [here](https://www.postgresql.org/docs/10/static/libpq-ssl.html#LIBPQ-SSL-PROTECTION)

**Comment by jacek-wojciechowski-dell** [comment]
@jonjomckay yeah... that totally looks like a bug in PostgreSQL JDBC driver...

worth testing if ''disable' works :) maybe we need to get rid of this param completly if SSL not enabled

**Comment by m-kuhn** [comment]
I tested a couple of parameters:

"require" and "disable" are supported
"prefer" and "allow" are not supported
